### ISSUE TITLE
Merge upstream 10.2

### ISSUE THREAD

**Issue by midenok** [tempesta-tech/mariadb#54]
- [x] Move fix in sql_yacc from part I to proper past commit;
- [x] Make timestamp from thd in trx not lost on next update;
- [x] Move typo fix in dict0load.cc:66 to proper commit.
- [x] Move fix  in row0mysql.cc:2728

```
            que_thr_stop_for_mysql_no_error(thr, trx);
```
- [x] Move warning fix (warning appears at _part IV_):

```
In file included from /home/midenok/src/mariadb/10.2/src/sql/sql_class.h:31:0,
                 from /home/midenok/src/mariadb/10.2/src/unittest/sql/explain_filename-t.cc:22:
/home/midenok/src/mariadb/10.2/src/sql/field.h: In constructor ‘Column_definition::Column_definition()’:
/home/midenok/src/mariadb/10.2/src/sql/field.h:3808:6: warning: ‘Column_definition::check_constraint’ will be initialized after [-Wreorder]
     *check_constraint;               // Check constraint
      ^
/home/midenok/src/mariadb/10.2/src/sql/field.h:3796:26: warning:   ‘Column_definition::enum_column_versioning Column_definition::versioning’ [-Wreorder]
   enum_column_versioning versioning;
                          ^
/home/midenok/src/mariadb/10.2/src/sql/field.h:3810:3: warning:   when initialized here [-Wreorder]
   Column_definition():
   ^
```
- [x] Fix full test results

**Comment by midenok** [comment]
~~Depends on #59.~~

**Comment by midenok** [comment]
~~Depends on #63.~~

**Comment by midenok** [comment]
~~Depends on #64.~~

**Comment by midenok** [comment]
~~Depends on #65.~~

**Comment by midenok** [comment]
~~Depends on #67.~~

**Comment by midenok** [comment]
Closed by 026d3276c14f1c70090dc4076cffc5b9f6c2812d
